src/core/read_halos-velociraptor.c

```diff
@@ -603,22 +321,24 @@ void read_trees__velociraptor_aug(int snapshot,
         // TODO: What masses and radii should I use for centrals (inclusive vs. exclusive etc.)?
         if (halo->Type == 0) {
           fof_group_t* fof_group = &fof_groups[*n_fof_groups];
-
-          if (tree_entry.Mass_200crit <= 0) {
-            // This "halo" is not above the virial threshold!  Use
-            // proxy masses, but flag this fact so we know not to do
-            // any or allow any hot halo to exist.
-            halo->TreeFlags |= TREE_CASE_BELOW_VIRIAL_THRESHOLD;
-            //fof_group->Mvir = tree_entry.Mass_FOF;
-            fof_group->Mvir = (double)tree_entry.Mass_tot * run_globals.params.Hubble_h * mass_unit_to_internal;
+          
+          // This check is to ensure sensible values of mass_200crit and avoid having 
+          // very weird halos
+          
+          if ((tree_entry.Mass_200crit < 5 * tree_entry.Mass_tot)) {
```

Comment:
i thought you've decided to remove this?

Reply:
Yes but I wanted to test the impact also on the augmented trees to be extra sure... Now that I saw that it works, I removed this check but leaving the comments